Gems/AudioSystem/Code/Source/Engine/AudioSystem.cpp

```diff
@@ -190,47 +181,58 @@ namespace Audio
         }
 
         m_apAudioProxiesToBeFreed.clear();
-
-    #if !defined(AUDIO_RELEASE)
-        DrawAudioDebugData();
-    #endif // !AUDIO_RELEASE
     }
 
     ///////////////////////////////////////////////////////////////////////////////////////////////////
     void CAudioSystem::InternalUpdate()
     {
         // Audio Thread!
+        AZ_Assert(g_audioThreadId == AZStd::this_thread::get_id(), "AudioSystem::InternalUpdate - called from non-Audio thread!");
         AZ_PROFILE_FUNCTION(Audio);
 
         auto startUpdateTime = AZStd::chrono::system_clock::now();        // stamp the start time
 
         bool handledBlockingRequests = false;
-        {
-            AZStd::lock_guard<AZStd::mutex> lock(m_blockingRequestsMutex);
-            handledBlockingRequests = ProcessRequests(m_blockingRequestsQueue);
+
+        {   // Process a single blocking request, if any, and release the semaphore on the main thread.
+            // This ensures that main thread will become unblocked quickly.
+            // If blocking requests were processed, can skip processing of normal requests and skip having
+            // the audio thread block through the rest of its update period.
+            AZStd::scoped_lock lock(m_blockingRequestsMutex);
+            if (!m_blockingRequestsQueue.empty())
+            {
+                // Blocking request...
+                AudioRequestVariant& request(m_blockingRequestsQueue.front());
+                m_oATL.ProcessRequest(AZStd::move(request));
+                m_blockingRequestsQueue.pop_front();
```

Comment:
If you swap the `pop_front` call with the `ProcessRequest` call, you can unlock the m_blockingRequestMutex before the call to ProcessRequests.